src/dbSta/src/dbNetwork.cc

```diff
@@ -82,6 +85,71 @@ char* tmpStringCopy(const char* str)
   return tmp;
 }
 
+//
+// Handling of object ids (Hierachy Mode)
+//--------------------------------------
+//
+// The database assigns a number to each object. These numbers
+// are scoped based on the type. Eg dbModInst 1..N or dbInst 1..N.
+// The timer requires a unique id for each object for its visit
+// pattern, so we uniquify the numbers by suffixing a discriminating
+// address pattern to the lower bits and shifting.
+// Everytime a new type of timing related object is added, we
+// must update this code, so it is isolated and marked up here.
+//
+// The id is used by the STA traversers to accumulate visited.
+// lower 4  bits used to encode type
+//
+
+#define DBITERM_ID 0x0
+#define DBBTERM_ID 0x1
+#define DBINST_ID 0x2
+#define DBNET_ID 0x3
+#define DBMODITERM_ID 0x4
+#define DBMODBTERM_ID 0x5
+#define DBMODINST_ID 0x6
+#define DBMODNET_ID 0x7
+#define DBMODULE_ID 0x8
+// Number of lower bits used
+#define DBIDTAG_WIDTH 0x4
+
+ObjectId getDbNwkObjectId(dbObjectType typ, ObjectId db_id)
+{
+  switch (typ) {
+    case dbITermObj: {
+      return ((db_id << DBIDTAG_WIDTH) | DBITERM_ID);
+    } break;
+    case dbBTermObj: {
+      return ((db_id << DBIDTAG_WIDTH) | DBBTERM_ID);
+    } break;
+    case dbInstObj: {
+      return ((db_id << DBIDTAG_WIDTH) | DBINST_ID);
+    } break;
+    case dbNetObj: {
+      return ((db_id << DBIDTAG_WIDTH) | DBNET_ID);
+    } break;
+    case dbModITermObj: {
+      return ((db_id << DBIDTAG_WIDTH) | DBMODITERM_ID);
+    } break;
+    case dbModBTermObj: {
+      return ((db_id << DBIDTAG_WIDTH) | DBMODBTERM_ID);
+    } break;
+    case dbModInstObj: {
+      return ((db_id << DBIDTAG_WIDTH) | DBMODINST_ID);
+    } break;
+    case dbModNetObj: {
+      return ((db_id << DBIDTAG_WIDTH) | DBMODNET_ID);
+    } break;
+    case dbModuleObj: {
+      return ((db_id << DBIDTAG_WIDTH) | DBMODULE_ID);
+    } break;
+    default:
+      return 0;
```

Comment:
Should we return an error or is 0 sometimes expected? I fear a silent corruption bug if a new object type is added in the future.

Reply:
There are many types not handled here.  Adding them all wouldn't add much clarity imho.